scripts/extract_timings.py

```diff
@@ -0,0 +1,39 @@
+#! /usr/bin/env python
+# Time-stamp: <2016-05-31 14:58:27 chrplr>
+#
+
+""" Extracts slice timing information from a dicom file.
+To be passed  as argument to slice_order for slice timing correction algorithms """
+
```

Comment:
On Mon, Jun 20, 2016 at 9:34 AM, Christophe Pallier <
notifications@github.com> wrote:

> In scripts/extract_timings.py
> <https://github.com/neurospin/pypreprocess/pull/231#discussion_r67646509>:
>
> > @@ -0,0 +1,39 @@
> > +#! /usr/bin/env python
> > +# Time-stamp: <2016-05-31 14:58:27 chrplr>
> > +#
> > +
> > +""" Extracts slice timing information from a dicom file.
> > +To be passed  as argument to slice_order for slice timing correction algorithms """
> > +
>
> I have just tried with a list of integers (slice numbers) for slice_order
> and got a crash:
>
> File "/home/cp983411/code/pypreprocess/pypreprocess/slice_timing.py", line
> 56, in get_slice_indices
> raise ValueError("Unknown slice order '%s'!" % slice_order)
> ValueError: Unknown slice order '1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17
> 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42
> 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67
> 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84'!
>
This is because the only allowed strings are "ascending" or "descending".
Otherwise it must be a list of integers. I'm updating the docstring for
this function to make this clearer. In summary, the logic for converting a
string of of integers (e.g "0 1 2 3 4") into a list of into a list of
integers ([0, 1, 2, 3, 4]) should be done out-side this function.

> It seems that the list is recognized a a string (isinstance(slice_order,
> basestring) is True), should I use a list syntax with '[' '],' and ','?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/neurospin/pypreprocess/pull/231/files/8caa896e8ea92a2478829bd93dbdbe53d287f26a#r67646509>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAms1G1As_V6SfsEg5e5RKBnQLsQMxl_ks5qNkKigaJpZM4IqgU1>
> .
>



-- 
Elvis Dohmatob,
*sent from my  potato...*
http://dohmatob.blogspot.fr/, https://team.inria.fr/parietal/elvis/,
https://github.com/dohmatob